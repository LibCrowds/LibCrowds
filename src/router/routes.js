import config from '@/config'
import store from '@/store'

import Core from '@/components/layouts/Core'
import Landing from '@/pages/Landing'

import About from '@/pages/pybossa/home/About'
import Data from '@/pages/pybossa/home/Data'
import TOS from '@/pages/pybossa/help/TOS'
import Privacy from '@/pages/pybossa/help/Privacy'
import Cookies from '@/pages/pybossa/help/Cookies'
import Api from '@/pages/pybossa/help/Api'

export default [

  // Core site
  { path: '/', name: 'core', component: Core },
  { path: 'tos', name: 'tos', component: TOS },
  { path: 'privacy', name: 'privacy', component: Privacy },
  { path: 'cookies', name: 'cookies', component: Cookies },
  { path: 'api', name: 'api', component: Api },

  // PYBOSSA meta-category sites
  {
    path: '/site/:shortname',
    component: {
      render (c) {
        return c('router-view')
      }
    },
    beforeEnter: (to, from, next) => {
      let siteConfig = config.sites.find(function (site) {
        return site.shortname === to.params.shortname
      })

      if (typeof siteConfig === 'undefined') {
        next({ name: '404' })
      }

      store.commit('SET_SITE_CONFIG', siteConfig)
      next()
    },
    children: [
      { path: '', name: 'home', component: Landing },
      { path: 'about', name: 'about', component: About },
      { path: 'data', name: 'data', component: Data }
    ]
  },

  // Errors
  {
    path: '/404',
    name: '404',
    alias: '*',
    component: require('@/pages/Error'),
    props: {
      title: 'Page Not Found',
      description: 'We could not find the page you were looking for'
    }
  }
]
