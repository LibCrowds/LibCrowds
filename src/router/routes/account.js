import CoreLayout from '@/layouts/Core'
import DashboardLayout from '@/layouts/Dashboard'

import Profile from '@/pages/account/Profile'
import ApiSettings from '@/pages/account/settings/ApiSettings'
import AvatarSettings from '@/pages/account/settings/AvatarSettings'
import ProfileSettings from '@/pages/account/settings/ProfileSettings'
import SecuritySettings from '@/pages/account/settings/SecuritySettings'
import Signin from '@/pages/account/Signin'
import Register from '@/pages/account/Register'
import ForgottenPassword from '@/pages/account/ForgottenPassword'
import ResetPassword from '@/pages/account/ResetPassword'
import Newsletter from '@/pages/account/Newsletter'
import Projects from '@/pages/account/Projects'

export default [
  {
    path: '/account',
    component: CoreLayout,
    beforeEnter: function (to, from, next) {
      if (to.fullPath === '/account') {
        next({ name: '404' })
      }
      next()
    },
    children: [
      {
        path: 'signin',
        name: 'signin',
        component: Signin
      },
      {
        path: 'register',
        name: 'register',
        component: Register
      },
      {
        path: 'forgot-password',
        name: 'forgot-password',
        component: ForgottenPassword
      },
      {
        path: 'reset-password',
        name: 'reset-password',
        component: ResetPassword
      },
      {
        path: 'newsletter',
        name: 'newsletter',
        component: Newsletter
      },
      {
        path: ':username',
        name: 'profile',
        component: Profile
      },
      {
        path: ':username/projects',
        name: 'account-projects',
        component: Projects
      },
      {
        path: 'register/confirmation',
        name: 'register-confirmation',
        beforeEnter (to, from, next) {
          pybossaApi.get('/account/register/confirmation', {
            params: to.query
          }).then(r => {
            next({ path: r.data.next })
          }).catch(err => {
            next({ name: String(err.response.status) })
          })
        }
      }
    ]
  },
  {
    path: '/account/:username/update',
    component: DashboardLayout,
    props: {
      dashboardNavItems: [
        {
          id: 'account-api-settings',
          label: 'API',
          link: {
            name: 'account-api-settings'
          }
        },
        {
          id: 'account-avatar-settings',
          label: 'Avatar',
          link: {
            name: 'account-avatar-settings'
          }
        },
        {
          id: 'account-profile-settings',
          label: 'Profile',
          link: {
            name: 'account-profile-settings'
          }
        },
        {
          id: 'account-security-settings',
          label: 'Security',
          link: {
            name: 'account-security-settings'
          }
        }
      ]
    },
    children: [
      {
        path: '/',
        name: 'account-settings',
        redirect: {
          name: 'account-profile-settings'
        }
      },
      {
        path: 'api',
        name: 'account-api-settings',
        component: ApiSettings
      },
      {
        path: 'avatar',
        name: 'account-avatar-settings',
        component: AvatarSettings
      },
      {
        path: 'profile',
        name: 'account-profile-settings',
        component: ProfileSettings
      },
      {
        path: 'security',
        name: 'account-security-settings',
        component: SecuritySettings
      }
    ]
  }
]
