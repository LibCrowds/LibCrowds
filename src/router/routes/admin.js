import DashboardLayout from '@/layouts/Dashboard'

import AdminDashboard from '@/pages/admin/AdminDashboard'
import Administrators from '@/pages/admin/Administrators'
import Publications from '@/pages/admin/Publications'
import Categories from '@/pages/admin/Categories'
import EditCategory from '@/pages/admin/EditCategory'
import Featured from '@/pages/admin/Featured'

export default [
  {
    path: '/admin',
    component: DashboardLayout,
    props: {
      dashboard: 'admin'
    },
    children: [
      {
        path: '/',
        name: 'admin-dashboard',
        component: AdminDashboard
      },
      {
        path: 'administrators',
        name: 'admin-administrators',
        component: Administrators
      },
      {
        path: 'announcement',
        name: 'admin-publications',
        component: Publications
      },
      {
        path: 'categories',
        name: 'admin-categories',
        component: Categories
      },
      {
        path: 'categories/update/:categoryid',
        name: 'admin-edit-category',
        component: EditCategory
      },
      {
        path: 'featured',
        name: 'admin-featured',
        component: Featured
      }
    ]
  }
]
