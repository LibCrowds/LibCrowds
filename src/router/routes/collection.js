import siteConfig from '@/siteConfig'
import CollectionBase from '@/pages/collection/Base'
import CollectionHome from '@/pages/collection/Home'
import CollectionAbout from '@/pages/collection/About'
import CollectionData from '@/pages/collection/Data'
// import CollectionResults from '@/pages/collection/Results'
import CollectionContribute from '@/pages/collection/Contribute'
import CollectionPresenter from '@/pages/collection/Presenter'

const routes = [
  {
    path: '/collection/:collectionname',
    component: CollectionBase,
    beforeEnter: function (to, from, next) {
      if (to.params.collectionname in siteConfig.collections) {
        next()
      } else {
        next({ name: '404' })
      }
    },
    children: [
      {
        path: '/',
        name: 'collection-home',
        component: CollectionHome
      },
      {
        path: 'about',
        name: 'collection-about',
        component: CollectionAbout
      },
      {
        path: 'data',
        name: 'collection-data',
        component: CollectionData
      },
      {
        path: 'contribute',
        name: 'collection-contribute',
        component: CollectionContribute
      },
      {
        path: 'project/:shortname/presenter',
        name: 'collection-presenter',
        component: CollectionPresenter
      }
    ]
  }
]

// Conditional routes (TODO: Update)
// if (config.resultsComponent) {
//   routes.push({
//     path: '/collection/:collectionname/statistics',
//     name: 'results',
//     component: Collection
//   })
// }

export default routes
