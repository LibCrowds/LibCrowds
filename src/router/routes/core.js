import AppBase from '@/pages/AppBase'
import Landing from '@/pages/core/Landing'
import Statistics from '@/pages/core/Statistics'

export default [
  {
    path: '/',
    component: AppBase,
    children: [
      {
        path: '/',
        name: 'landing',
        component: Landing
      },
      {
        path: 'statistics',
        name: 'statistics',
        component: Statistics
      }
    ]
  }
]
