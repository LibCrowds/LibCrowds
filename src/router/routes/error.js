import Error from '@/pages/Error'

export default [
  {
    path: '/404',
    name: '404',
    alias: '*',
    component: Error,
    props: {
      title: 'Page Not Found',
      description: 'We could not find the page you were looking for'
    }
  }
]
