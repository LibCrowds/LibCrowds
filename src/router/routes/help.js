import DashboardLayout from '@/layouts/Dashboard'

import API from '@/pages/help/API'
import Cookies from '@/pages/help/Cookies'
import Privacy from '@/pages/help/Privacy'
import TOS from '@/pages/help/TOS'

export default [
  {
    path: '/help',
    component: DashboardLayout,
    props: {
      dashboardNavItems: [
        {
          id: 'help-api',
          label: 'API',
          link: {
            name: 'help-api'
          }
        },
        {
          id: 'help-cookies',
          label: 'Cookies Policy',
          link: {
            name: 'help-cookies'
          }
        },
        {
          id: 'help-privacy',
          label: 'Privacy Policy',
          link: {
            name: 'help-privacy'
          }
        },
        {
          id: 'help-tos',
          label: 'Terms of Service',
          link: {
            name: 'help-tos'
          }
        }
      ]
    },
    children: [
      {
        path: '/api',
        name: 'help-api',
        component: API
      },
      {
        path: '/cookies',
        name: 'help-cookies',
        component: Cookies
      },
      {
        path: '/privacy',
        name: 'help-privacy',
        component: Privacy
      },
      {
        path: '/tos',
        name: 'help-tos',
        component: TOS
      }
    ]
  }
]
