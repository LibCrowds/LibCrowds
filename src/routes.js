import config from '@/config'

export default [
  {
    path: '/',
    component: require('@/pages/Landing')
  },
  {
    path: '/site/:sitename',
    component: require('@/components/layouts/Main'),
    beforeEnter: (to, from, next) => {
      if (!(to.params.sitename in config.pybossaSites)) {
        next({ name: '404' })
      }
      next()
    },
    children: [
      {
        path: '',
        name: 'home',
        component: require('@/pages/Landing')
      },
      {
        path: 'about',
        name: 'about',
        component: require('@/pages/pybossa/home/About')
      },
      {
        path: 'projects',
        name: 'projects',
        component: require('@/pages/pybossa/projects/Projects')
      },
      {
        path: 'stats',
        name: 'stats',
        component: require('@/pages/pybossa/home/Stats')
      },
      {
        path: 'results',
        name: 'results',
        component: require('@/pages/pybossa/home/Results')
      },
      {
        path: 'data',
        name: 'data',
        component: require('@/pages/pybossa/home/About')
      }
    ]
  },
  {
    path: '/404',
    name: '404',
    alias: '*',
    component: require('@/pages/Error'),
    meta: { title: 'Page Not Found' }
  }
]
