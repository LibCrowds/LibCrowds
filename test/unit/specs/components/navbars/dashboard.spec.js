import VueRouter from 'vue-router'
import BootstrapVue from 'bootstrap-vue'
import NuxtLink from '@/.nuxt/components/nuxt-link'

import { routes } from '@/test/fixtures/routes'
import { mount, createLocalVue } from 'vue-test-utils'
import DashboardNavbar from '@/components/navbars/Dashboard'

describe('Dashboard navbar', () => {
  let localVue = null
  let wrapper = null
  let router = null
  let navItems = null

  beforeEach(() => {
    localVue = createLocalVue()
    localVue.use(BootstrapVue)
    localVue.use(VueRouter)
    localVue.component(NuxtLink.name, NuxtLink)
    router = new VueRouter({
      routes
    })
    navItems = [
      {
        label: 'Dashboard',
        link: {
          name: 'admin-site-dashboard'
        }
      },
      {
        label: 'Users',
        link: {
          name: 'admin-site-users'
        }
      }
    ]
  })

  it('renders correctly in top position', () => {
    wrapper = mount(DashboardNavbar, {
      localVue,
      router,
      propsData: {
        position: 'top',
        navItems: navItems
      }
    })
    const renderer = require('vue-server-renderer').createRenderer()
    renderer.renderToString(wrapper.vm, (err, str) => {
      expect(str).toMatchSnapshot()
    })
  })

  it('renders correctly in side position', () => {
    wrapper = mount(DashboardNavbar, {
      localVue,
      router,
      propsData: {
        position: 'side',
        navItems: navItems
      }
    })
    const renderer = require('vue-server-renderer').createRenderer()
    renderer.renderToString(wrapper.vm, (err, str) => {
      expect(str).toMatchSnapshot()
    })
  })
})
