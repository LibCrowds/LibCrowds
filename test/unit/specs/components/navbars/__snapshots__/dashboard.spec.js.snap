// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard navbar renders correctly in side position 1`] = `

<div id="dashboard-navbar"
     data-server-rendered="true"
>
  <nav class="side">
    <div class="brand-wrapper navbar-light text-center">
      <nuxt-link to="[object Object]"
                 class="brand navbar-brand"
      >
        My Brand
      </nuxt-link>
    </div>
    <ul class="nav">
      <li data-index="0"
          class="nav-item"
      >
        <a href="#"
           target="_self"
           aria-disabled="false"
           class="nav-link"
        >
          Dashboard
        </a>
      </li>
      <li data-index="1"
          class="nav-item"
      >
        <a href="#"
           target="_self"
           aria-disabled="false"
           class="nav-link"
        >
          Featured Projects
        </a>
      </li>
    </ul>
  </nav>
</div>

`;

exports[`Dashboard navbar renders correctly in top position 1`] = `

<div id="dashboard-navbar"
     data-server-rendered="true"
>
  <nav class="top">
    <div class="brand-wrapper navbar-light text-center">
      <nuxt-link to="[object Object]"
                 class="brand navbar-brand"
      >
        My Brand
      </nuxt-link>
    </div>
    <ul class="nav">
      <li data-index="0"
          class="nav-item"
      >
        <a href="#"
           target="_self"
           aria-disabled="false"
           class="nav-link"
        >
          Dashboard
        </a>
      </li>
      <li data-index="1"
          class="nav-item"
      >
        <a href="#"
           target="_self"
           aria-disabled="false"
           class="nav-link"
        >
          Featured Projects
        </a>
      </li>
    </ul>
  </nav>
</div>

`;
